### PR TITLE
Introduce enum class

### PR DESCRIPTION
**Pull request by getand**
I've introduced a nicer interface to deal with enums.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -164,14 +164,18 @@ no value for the key.
 
 ```ruby
 class Book
-  STATES = NxtSupport::EnumHash['draft', 'revised', 'Published', 'in weird State']
+  STATES = NxtSupport::EnumCollection['draft', 'revised', 'Published', 'in weird State']
```
Comment (nsommer) [inline README.md]: Why `Collection`? Isn't this implicitly expressed by `Enum`?

Comment (getand) [reply README.md]: Well I've already used `Enum` as a class. Could be `Enums` I guess.

Comment (getand) [reply README.md]: @nsommer Can we proceed with this?

**Review by the-wendell — APPROVED** [review]

**Review by nsommer — APPROVED** [review]
Looks great (CHANGELOG and version are missing afaik) 👍

**Pull request merged**